core/nuxeo-page-provider.js

```diff
@@ -433,21 +433,20 @@ import './nuxeo-resource.js';
         params.queryParams = this.params.queryParams;
       } else if (Array.isArray(this.params)) {
         params.queryParams = this.params;
-      } else {
-        const namedParams = {};
-        Object.keys(this.params).forEach((key) => {
-          let value = this.params[key];
-          if (value != null) {
-            if (Array.isArray(value)) {
-              value = JSON.stringify(value.map((item) => (item['entity-type'] ? item.uid || item.id : item)));
-            } else if (typeof value !== 'string') {
-              value = value['entity-type'] ? value.uid || value.id : JSON.stringify(value);
-            }
-            namedParams[key] = value;
-          }
-        }, this);
-        params.namedParameters = namedParams;
       }
+      const namedParams = {};
+      Object.keys(this.params).forEach((key) => {
```

Comment:
Don't think we want to run this if `Array.isArray(this.params)`. Also, we have above an `if(this.params.queryParams)` but now we always iterate on `this.params` keys, so we will handle the `queryParams` param twice, right? We could simply add an exception for `this.params.queryParams` on the loop. Something along these lines:
```
if (Array.isArray(this.params)) {
  params.queryParams = this.params;
} else {
  const namedParams = {};
  Object.keys(this.params).forEach((key) => {
    // do something if key === 'queryParams'
    ...
    }
  }, this);
  params.namedParameters = namedParams;
}
```

Reply:
👍  yes, plz don't forget to handle the `this.params.queryParams` use case. I guess the easiest path is to just to a `let queryParams;` before the ifs and then set it properly, to avoid changing `this.params` in the `forEach`.
this way you can just do a  `params.queryParams = queryParams` afterwards like we do with `namedParams`